catalog/dags/maintenance/trim_and_deduplicate_tags.py

```diff
@@ -0,0 +1,88 @@
+"""
+See the issue for context and motivation: https://github.com/WordPress/openverse/issues/4199
+
+This DAG triggers a run of the batched update DAG. It generates a new list of tags by
+trimming all existing tags and re-inserting only the distinct tags of the resulting list of tags.
+"""
+
+from datetime import datetime, timedelta
+from textwrap import dedent
+
+from airflow.decorators import dag
+from airflow.operators.trigger_dagrun import TriggerDagRunOperator
+
+from common.constants import DAG_DEFAULT_ARGS, MEDIA_TYPES
+from database.batched_update.constants import DAG_ID as BATCHED_UPDATE_DAG_ID
+
+
+DAG_ID = "trim_and_deduplicate_tags"
+
+
+@dag(
+    dag_id=DAG_ID,
+    schedule=None,
+    start_date=datetime(2024, 6, 3),
+    tags=["database"],
+    doc_md=__doc__,
+    max_active_runs=1,
+    default_args=DAG_DEFAULT_ARGS,
+)
+def trim_and_deduplicate_tags():
+    TriggerDagRunOperator.partial(
+        task_id=DAG_ID,
+        trigger_dag_id=BATCHED_UPDATE_DAG_ID,
+        wait_for_completion=True,
+        execution_timeout=timedelta(hours=5),
+        max_active_tis_per_dag=1,
+        map_index_template="""{{ task.conf['table_name'] }}""",
+        retries=0,
+    ).expand(
+        conf=[
+            {
+                "query_id": f"{DAG_ID}_{media_type}",
+                "table_name": media_type,
+                "select_query": (
+                    "WHERE "
+                    + (
+                        ""
+                        if media_type == "audio"
+                        # We've identified these image providers in particular definitely have tags that need trimming, so we can narrow the query starting there at least
+                        else (
+                            "provider IN ('flickr', 'geographorguk', 'thingiverse', 'animaldiversity', 'clevelandmuseum', 'digitaltmuseum') AND "
+                        )
+                    )
+                    + (
+                        f"""EXISTS (SELECT 1 FROM jsonb_array_elements({media_type}.tags) AS t WHERE t->>'name' ILIKE ' %' OR t->>'name' ILIKE '% ')"""
+                    )
+                ),
+                "update_query": (
+                    "SET updated_on = now(), "
```

Comment:
Huh. _Wow_, good catch. This is explicitly set during ingestion, rather than using a trigger. I suspect this has gone uncaught in all previous batched updates, including the popularity refreshes. I'll make a PR to fix popularity refreshes, and we may need to think about adding a trigger or else enforcing it in the batched update DAG somehow.

Reply:
The `add_license_url` DAG includes it, which I was using as a reference for some of this work. I guess you have something like this in mind, with respect to a trigger?

https://stackoverflow.com/questions/9556474/automatically-populate-a-timestamp-field-in-postgresql-when-a-new-row-is-inserte/9556527#9556527

It would seem non-controversial that if a row is updated the `updated_on` should change, and a trigger would enforce that, and easily enough too.

Reply:
I actually raised this in the recent media properties PR for the catalog! https://github.com/WordPress/openverse/pull/4366#issuecomment-2136191545

I can go ahead and make an issue for adding a trigger for this, given you're on board Staci.

Reply:
Agreed! Trigger is the right solution IMO, but I opened https://github.com/WordPress/openverse/pull/4460 to fix it in the popularity refresh in the interim. If we want to just jump ahead and go straight to adding the trigger instead that is also 100% fine with me.

Reply:
Issue created: https://github.com/WordPress/openverse/issues/4520